src/utils/version-patch.js

```diff
@@ -0,0 +1,76 @@
+/**
+ * This source file is part of the Swift.org open source project
+ *
+ * Copyright (c) 2021 Apple Inc. and the Swift project authors
+ * Licensed under Apache License v2.0 with Runtime Library Exception
+ *
+ * See https://swift.org/LICENSE.txt for license information
+ * See https://swift.org/CONTRIBUTORS.txt for Swift project authors
+*/
+/* eslint-disable no-plusplus */
+import { apply } from './json-patch';
+import { clone } from './data';
+
+/**
+ * Checks if a JSON file (typically index.json or a RenderNode) has version data
+ * @param  {JSON} data to be inspected for version data
+ * @returns {Boolean}
+ */
+function dataHasVersion(data) {
+  if (!data) return false;
+  if (!('metadata' in data)) return false;
+  if (!('version' in data.metadata)) return false;
+  return true;
+}
+/**
+ * Find the index of a specific Display Name in the version list
+ * @param  {string} displayName
+ * @param  {JSON} data
+ * @returns {number} index of display name, -1 if not found.
+ */
+function getDiffIndex(displayName, data) {
+  if (!dataHasVersion(data)) return -1;
+  if (!('versions' in data)) return -1;
+  const index = data.versions.findIndex(item => item.version.displayName === displayName);
+  return index;
+}
+/**
+ * Initialize a list of versions for a given JSON file
+ * @param  {JSON} data
+ * @returns {Array} of versions
+ */
+function initializeVersionList(data) {
+  if (dataHasVersion(data)) {
+    let versions = [];
+    if (data.versions) {
+      versions = data.versions.map(x => x.version.displayName);
+    }
+    versions.unshift(data.metadata.version.displayName);
```

Comment:
in case data doesnt have versions, why would you unshift an empty object?
I believe what you want to do looks more like this?

```js
function initializeVersionList(data) {
  if (!dataHasVersion(data) || data.versions) return [];
  const versionList = data.versions.map(x => x.version.displayName);
  versionList.unshift(data.metadata.version.displayName);
  return versionList;
}
```